docs/api/README.md

```diff
@@ -792,6 +793,35 @@ will be a single keyed object `{cancel: action}`, where action is the dispatched
 
 When resolving a `race`, the middleware automatically cancels all the losing Effects.
 
+### `race([...effects]) (with Array)`
+
+The same as [`race(effects)`](#raceeffects) but let's you to pass in an array of effects.
+
+#### Example
+
+The following example runs a race between two effects:
+
+1. A call to a function `fetchUsers` which returns a Promise
+2. A `CANCEL_FETCH` action which may be eventually dispatched on the Store
+
+```javascript
+import { take, call, race } from `redux-saga/effects`
+import fetchUsers from './path/to/fetchUsers'
+
+function* fetchUsersSaga {
+  const result = yield race([
+    call(fetchUsers),
+    take(CANCEL_FETCH)
+  ])
+}
+```
+
+If `call(fetchUsers)` resolves (or rejects) first, the result of `race` will be an object
+with a single keyed object `{'0': result}` where `result` is the resolved result of `fetchUsers`.
```

Comment:
Hm, this has realized me that even though `Object.keys` indeed allows us to pass in an array, the race's outcome is not an array in such case, but an object - so i.e. it cannot be destructured. If we want to put it into the docs, we should convert the result to an array.

Would you like to add this to the PR and adjust the proposed docs? Conversion can be made simply by adding a length property to the object and calling `[].slice.call(result)`. The whole work would needs to be done in [this function](https://github.com/redux-saga/redux-saga/blob/f609e805a7a877fa6b8c14934ad207a05df7aad2/src/internal/proc.js#L619) and should be really straightforward to implement. If you want to do it, please do not forget about adding tests for this :)

Reply:
No problem, indeed returning Array would be more consistent :) I could try to fix it myself, but I find it hard to understand how to return Array not Object in this function, this is not a pure function and no object is returned here, so I guess I need to follow this project's convention, but I need a small hint.

Btw, wouldn't `Array.from({ ...response, length: keys.length }) `be more legible than `[].slice.call(result)`?

Reply:
Result is returned through cb. Array.from would be better but its a part of es6 standard and we cant require everyone to polyfill it

Reply:
Ok thx. I did it but I have a small problem. In below test
```javascript
test('processor race between array of effects handling', assert => {
  assert.plan(1)

  let actual = []
  const timeout = deferred()
  const input = cb => {
    Promise.resolve(1).then(() => timeout.resolve(1)).then(() => cb({ type: 'action' }))
    return () => {}
  }

  function* genFn() {
    actual.push(
      yield io.race([
        io.take('action'),
        timeout.promise,
      ]),
    )
  }

  proc(genFn(), input).done.catch(err => assert.fail(err))

  const expected = [[, 1]]

  setTimeout(() => {
    assert.deepEqual(actual, expected, 'processor must fullfill race between array of effects')
    assert.end()
  })
})
```
I have problem with `const expected = [[, 1]]`. It is not the same as `[[undefined, 1]` and it breaks `eslint` rule: `error  Unexpected comma in middle of array  no-sparse-arrays`. Could we block this rule for this test or you have some alternative?

Reply:
Thanks for taking the time to work on it. Disabling the rule for this line is an option, you can do it with `// eslint-disable-next-line no-sparse-arrays`

Once you push it out im gonna checkout ur branch and why `assert.deepEqual` fails when passing `[[undefined, 1]]`

Reply:
@Andarist Any news on this? Btw, we can solve our our issue with `[...[, 1]]`, which will be converted to `[[undefined, 1]]`

Reply:
@klis87 Sorry for the long wait, I was really busy lately. I'm going atm through the issues and PRs, expect this to be handled soon.